esphome/components/script/__init__.py

```
@@ -29,6 +31,26 @@
     CONF_PARALLEL: ParallelScript,
 }
 
+# TODO add support for entity ids as params
+SCRIPT_PARAMS_NATIVE_TYPES = {
+    "bool": bool,
+    "int": cg.int32,
+    "float": float,
+    "string": cg.std_string,
+    "bool[]": cg.std_vector.template(bool),
+    "int[]": cg.std_vector.template(cg.int32),
+    "float[]": cg.std_vector.template(float),
+    "string[]": cg.std_vector.template(cg.std_string),
+}
+
+
+def get_script(script_id):
+    scripts = CORE.config.get(CONF_SCRIPT, {})
+    for script in scripts:
+        if script.get(CONF_ID, None) == script_id:
+            return script
+    return {}
```

This should probably raise an error if the script does not exist.

---

The config validation should have caught the error (referring to a non-existent id) before it gets here, so it (should) never reaches that last return line. Should I replace it with a raise still?

---

Yeah I think that would be better

---

I agree. For the unsuspecting reader, the fact that there is a return statement at that point, suggests that using a non-existent script id would be a valid case. Also, an exception would make the code stand on its own legs, instead of depending on config checking rules.

---

I've raised an exception now.